master/buildbot/process/workerforbuilder.py

```diff
@@ -14,16 +14,19 @@
 # Copyright Buildbot Team Members
 from twisted.internet import defer
 from twisted.python import log
+from twisted.python.constants import NamedConstant
+from twisted.python.constants import Names
 
 from buildbot.worker_transition import WorkerAPICompatMixin
 
-(ATTACHING,  # worker attached, still checking hostinfo/etc
- IDLE,  # idle, available for use
- PINGING,  # build about to start, making sure it is still alive
- BUILDING,  # build is running
- LATENT,  # latent worker is not substantiated; similar to idle
- SUBSTANTIATING,
- ) = range(6)
+
+class States(Names):
+    ATTACHING = NamedConstant()  # worker attached, still checking hostinfo/etc
+    IDLE = NamedConstant()  # idle, available for use
+    PINGING = NamedConstant()  # build about to start, making sure it is still alive
+    BUILDING = NamedConstant()  # build is running
+    LATENT = NamedConstant()  # latent worker is not substantiated; similar to idle
+    SUBSTANTIATING = NamedConstant()
```

Comment:
Mostly though, the reason I used `twisted.python.constants` is that is what I'm familiar with. But being in the standard library doesn't necessarily make a choice better (see the pointer to requests [here](https://docs.python.org/2/library/urllib2.html) for example).